### PR TITLE
Implement rule "module-required-files"

### LINKED ISSUE #4: Create rule "module-required-files"
Check if modules have the specified files. Can be strict or not.

.monolinter.json:

* Enable rule and configure it for requiring file "README.md" and "Makefile"

```
{
  "rules": {
    "module-required-files": {
      "strict": false,
      "files": ["README.md", "Makefile"]
     }
  }
}
```

### PR DESCRIPTION
**Pull request by tiagostutz**
Besides the implementation of the `module-required-files`, some refactors were done to better isolate the test cases.
* Add one more level for the test folders to allow specific folder/file compositions to be created per rule being tested
* Implement module-required-files rule with tests
* Created specific `Config` types for `module-required-files` rule and for `package-json-name` rule with proper refactor/casting

Closes #4

### CODE REVIEW

File: rules.md [hunk 1]
```diff
@@ -4,6 +4,40 @@ See below all rules that can be used for monorepo linting.
 
 Those configurations should be added to a file in the root of the monorepo called '.monolint.json'. If you create this file in intermediate folder (or even in the module folder), it will be merged to the root and default configurations also.
 
+## **module-required-files**
```
Comment (flaviostutz) [inline rules.md:7]: It seems the pipeline is failing because this file needs to be updated

File: src/types/Config.ts [hunk 2]
```diff
@@ -1,5 +1,17 @@
-export type Config = {
+type Config = {
   'module-markers'?: string[];
   'use-gitignore': boolean;
-  rules?: Record<string, boolean | Record<string, any>>;
+  rules?: Record<string, boolean | string | ConfigModuleRequiredFiles | ConfigPackageJsonSameName>;
 };
+
+type ConfigModuleRequiredFiles = {
```
Comment (flaviostutz) [inline src/types/Config.ts:7]: Nice!
We have to keep in mind that the number of typed here for the different rules will grow a lot

**Review by flaviostutz — CHANGES_REQUESTED** [review]
Nice work! I loved the refactorings! Thanks! Please check my comments

File: src/rules/packagejson-same-name.ts [hunk 3]
```diff
@@ -54,7 +56,7 @@ const rule: Rule = {
         results.push({
           valid: false,
           resource: packageFile,
-          message: `Couldn't load json file: ${err}`,
+          message: "Couldn't load file",
```
[the remark this answers is not in the document]
Comment (tiagostutz) [reply src/rules/packagejson-same-name.ts:59]: Fixed! 
Tbh I don't remember when/why I have changed it.

File: Makefile [hunk 4]
```diff
@@ -3,7 +3,8 @@ build:
 	npm run build
 
 run:
-	npx ts-node src/main.ts --base-dir=./src/rules/test-monorepo --verbose
+	# npx ts-node src/main.ts --base-dir=./src/rules/test-cases/general --verbose
```
[the remark this answers is not in the document]
Comment (tiagostutz) [reply Makefile:6]: Good catch!

File: src/types/Config.ts (hunk 2 above)
Comment (tiagostutz) [reply src/types/Config.ts:7]: Yep, I was thinking about that. Probably we will have to think in a way to group things, like "module related types" on a file and so on...

File: rules.md (hunk 1 above)
Comment (tiagostutz) [reply rules.md]: Done! Thanks.!

**Review by flaviostutz — APPROVED** [review]
Wow! On top of the new rule implementation we had really good improvements to our overall code base! Thanks!

**Pull request merged**